### PR TITLE
fixes route name annoucement in navigator

### LINKED ISSUE #50027: Named routes disappear after refreshing
Refreshing the page replaces the initial route in the URL bar.

### Steps to reproduce:

1. Run the example app in Chrome
2. click the button
3. Refresh the browser
4. Observe that the /page2 in the URL bar goes away.

<details><summary>Example app</summary>
<p>

```python
import 'package:flutter/material.dart';

void main() {
  runApp(DeepLinkingApp());
}

class DeepLinkingApp extends StatefulWidget {
  @override
  _DeepLinkingAppState createState() => _DeepLinkingAppState();
}

class _DeepLinkingAppState extends State<DeepLinkingApp> {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      routes: {
        '/': (context) => Page1(),
        '/page2': (context) => Page2(),
      },
      initialRoute: '/',
    );
  }
}

class Page1 extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: RaisedButton(
          onPressed: () {
            Navigator.of(context).pushNamed('/page2');
          },
          child: Text('Go to Page 2'),
        ),
      ),
    );
  }
}

class Page2 extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Text('Page 2'),
      ),
    );
  }
}

```

</p>
</details>

routing

### PR DESCRIPTION
**Pull request by chunhtai**
## Description

Due to recent refactoring work on navigator, a bug was introduced in route announcement. Since the current api will also be outdated after the page api is implemented, I would like to refactor the route announcement while fixing this bug. 
This PR requires Engine pr https://github.com/flutter/engine/pull/16523.

## Related Issues
Closes https://github.com/flutter/flutter/issues/50027

## Tests

I added the following tests:

Existing test coverage

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by mdebbar — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/route_notification_messages.dart [hunk 1]
```diff
@@ -9,33 +9,29 @@ import 'navigator.dart';
```
Comment (mdebbar) [inline packages/flutter/lib/src/widgets/route_notification_messages.dart]: The analyzer is complaining that the above import (`import 'navigator.dart;`) isn't used.

**Comment by chunhtai** [comment]
This is currently wait for engine roll to landed in framework

**Comment by mdebbar** [comment]
@chunhtai FYI engine roll landed successfully https://github.com/flutter/flutter/pull/50535.

**Pull request merged**